File: app/src/main/java/de/bahnhoefe/deutschlands/bahnhofsfotos/util/Timetable.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.util;

import android.content.Intent;
import android.net.Uri;

import androidx.annotation.Nullable;

import org.apache.commons.lang3.StringUtils;

import de.bahnhoefe.deutschlands.bahnhofsfotos.model.Country;
import de.bahnhoefe.deutschlands.bahnhofsfotos.model.Station;

public class Timetable {

    /**
     * Build an intent for an action to view a timetable for the station.
     *
     * @return the PendingIntent built.
     */
    @Nullable
    public Intent createTimetableIntent(final Country country, final Station station) {
        if (!country.hasTimetableUrlTemplate()) {
            return null;
        }

        var timeTableTemplate = country.getTimetableUrlTemplate();
        timeTableTemplate = timeTableTemplate.replace("{id}", station.getId());
        timeTableTemplate = timeTableTemplate.replace("{title}", station.getTitle());
        timeTableTemplate = timeTableTemplate.replace("{DS100}", StringUtils.trimToEmpty(station.getDs100()));

        final var timetableIntent = new Intent(Intent.ACTION_VIEW);
        timetableIntent.setData(Uri.parse(timeTableTemplate));
        return timetableIntent;
    }
}

File: app/src/main/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/util/Timetable.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.util

import android.content.Intent
import android.net.Uri
import de.bahnhoefe.deutschlands.bahnhofsfotos.model.Country
import de.bahnhoefe.deutschlands.bahnhofsfotos.model.Station
import org.apache.commons.lang3.StringUtils

class Timetable {
    /**
     * Build an intent for an action to view a timetable for the station.
     *
     * @return the PendingIntent built.
     */
    fun createTimetableIntent(country: Country, station: Station?): Intent? {
        if (!country.hasTimetableUrlTemplate()) {
            return null
        }
        var timeTableTemplate = country.timetableUrlTemplate
        timeTableTemplate = timeTableTemplate!!.replace("{id}", station!!.id!!)
        timeTableTemplate = timeTableTemplate.replace("{title}", station.title!!)
        timeTableTemplate = timeTableTemplate.replace(
            "{DS100}", StringUtils.trimToEmpty(
                station.ds100
            )
        )
        val timetableIntent = Intent(Intent.ACTION_VIEW)
        timetableIntent.data = Uri.parse(timeTableTemplate)
        return timetableIntent
    }
}